File: mobile/src/main/java/com/bayapps/android/robophish/ui/ShowPagerAdapter.java
package com.bayapps.android.robophish.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.viewpager.widget.PagerAdapter;

import com.bayapps.android.robophish.R;

/**
 * Created by mikevoyt on 8/11/16.
 */

public class ShowPagerAdapter extends PagerAdapter {

    private LayoutInflater mInflater;
    private View mRootView;

    public ShowPagerAdapter(LayoutInflater inflater, View rootView) {
        mInflater = inflater;
        mRootView = rootView;
    }


    private static boolean mAdded = false;
    @Override
    public Object instantiateItem(ViewGroup collection, int position) {

        View view = null;
        if (position == 0) view = mRootView.findViewById(R.id.tracks);
        if (position == 1) view = mRootView.findViewById(R.id.setlist);
        if (position == 2) view = mRootView.findViewById(R.id.reviews);
        if (position == 3) view = mRootView.findViewById(R.id.tapernotes);
        collection.addView(view);

        return view;
    }


    @Override
    public void destroyItem(ViewGroup collection, int position, Object view) {
        collection.removeView((View) view);
    }


    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0: return "Tracks";
            case 1: return "Setlist";
            case 2: return "Reviews";
            case 3: return "Taper Notes";
        }
        return null;
    }

}

File: mobile/src/main/java/com/bayapps/android/robophish/ui/ShowPagerAdapter.kt
package com.bayapps.android.robophish.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import com.bayapps.android.robophish.R

class ShowPagerAdapter(
    private val onTracksViewCreated: (View) -> Unit,
    private val onSetlistViewCreated: (View) -> Unit,
    private val onReviewsViewCreated: (View) -> Unit,
    private val onTaperNotesViewCreated: (View) -> Unit
) : RecyclerView.Adapter<ShowPagerAdapter.PageViewHolder>() {

    private val pages = listOf(
        Page("Tracks", R.layout.fragment_show_tracks),
        Page("Setlist", R.layout.fragment_show_setlist),
        Page("Reviews", R.layout.fragment_show_reviews),
        Page("Taper Notes", R.layout.fragment_show_tapernotes)
    )

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PageViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
        return PageViewHolder(view)
    }

    override fun onBindViewHolder(holder: PageViewHolder, position: Int) {
        when (position) {
            0 -> onTracksViewCreated(holder.itemView)
            1 -> onSetlistViewCreated(holder.itemView)
            2 -> onReviewsViewCreated(holder.itemView)
            3 -> onTaperNotesViewCreated(holder.itemView)
        }
    }

    override fun getItemViewType(position: Int): Int = pages[position].layoutId

    override fun getItemCount(): Int = pages.size

    fun getPageTitle(position: Int): CharSequence = pages[position].title

    class PageViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    data class Page(val title: String, @LayoutRes val layoutId: Int)
}
